features/keychain/module/crypto/tweak.js

```diff
@@ -0,0 +1,24 @@
+import ecc from '@exodus/bitcoinerlab-secp256k1'
+import { utils } from '@noble/secp256k1'
+
+const tapTweakHash = (publicKey, h) => {
+  const xOnlyPoint = ecc.xOnlyPointFromPoint(publicKey)
+  const hash = utils.taggedHashSync('TapTweak', Buffer.concat(h ? [xOnlyPoint, h] : [xOnlyPoint]))
+  return Buffer.from(hash)
+}
+
+export const tweakPrivateKey = ({ hdkey, tweakOptions = {} }) => {
+  const { publicKey } = hdkey
+  const { tweakHash } = tweakOptions
+
+  let privateKey = publicKey[0] === 3 ? ecc.privateNegate(hdkey.privateKey) : hdkey.privateKey
+  privateKey = Buffer.from(ecc.privateAdd(privateKey, tapTweakHash(publicKey, tweakHash)))
```

Comment:
Can we explicitly check that the point is compressed? `publicKey[0] === 3` only works on compressed public keys.

Reply:
I would appreciate a code snippet.

Reply:
https://github.com/ExodusMovement/exodus-oss/pull/90/commits/0b453f28edde20865cbbcdef04e0f0d5eecef9f0